app/src/main/java/org/schabi/newpipe/fragments/MainFragment.java

```diff
@@ -120,8 +117,9 @@ public void onResume() {
     public void onDestroy() {
         super.onDestroy();
         tabsManager.unsetSavedTabsListener();
-        if (viewPager != null) {
-            viewPager.setAdapter(null);
+        if (binding != null) {
+            binding.pager.setAdapter(null);
+            binding = null;
```

Comment:
Why are you setting `binding = null` here and not in `onDestroyView` like all other fragments?

<sub>(unrelated to this PR, just a note for the future) `pagerAdapter` should be set to `null` here, as LeakCanary reports it as leaking.</sub>

Reply:
Because the pager adapter is being set to null in that method, and the `onDestroyView` method is called before `onDestroy`: https://developer.android.com/guide/fragments/lifecycle.

Reply:
That logic should be in onDestroyView, since the view gets re-created all the time.

onDestroy happens when the fragment is killed, what does that mean?
onDestroyView is called everytime whilst onDestroy isn't.

You should move it to onDestroyView

Reply:
Okay.